erpnext/compliance/doctype/compliance_info/compliance_info.py

```diff
@@ -59,28 +59,39 @@ def get_existing_licensees(license_number, party_type):
 
 def validate_license_expiry(doc):
 	if doc.doctype in ("Sales Order", "Sales Invoice", "Delivery Note"):
-		validate_entity_license("Customer", doc.customer)
+		validate_entity_license(doc, "Customer", doc.customer)
 	elif doc.doctype in ("Supplier Quotation", "Purchase Order", "Purchase Invoice", "Purchase Receipt"):
-		validate_entity_license("Supplier", doc.supplier)
+		validate_entity_license(doc, "Supplier", doc.supplier)
 	elif doc.doctype == "Quotation" and doc.quotation_to == "Customer":
-		validate_entity_license("Customer", doc.party_name)
+		validate_entity_license(doc, "Customer", doc.party_name)
 
 
 @frappe.whitelist()
-def validate_entity_license(party_type, party_name):
+def validate_entity_license(doc, party_type, party_name):
+	is_compliance_item = False
+	compliance_items = frappe.get_all('Compliance Item', fields=['item_code'])
+	if not compliance_items:
+		return
+
+	for item in (doc.get("items") or []):
+		compliance_item = next((data for data in compliance_items if data.get("item_code") == item.get("item_code")), None)
+		if compliance_item:
+			is_compliance_item = True
```

Comment:
Create a new whitelisted function called `validate_doc_license` and handle this there

The `validate_entity_license` function should do just one thing, and that is to verify the party's license number